src/back/spv/writer.rs

```diff
@@ -1666,8 +1745,11 @@ impl Writer {
             }
         }
 
-        function.blocks.push(block);
-        Ok(id)
+        if block.termination.is_none() {
+            block.termination = Some(super::instructions::instruction_branch(merge_id.unwrap()));
```

Comment:
does it make sense to keep `merge_id` optional if we are unwrapping it anyway?

Reply:
Yes, because the "last" root block won't have a `merge_id`, as it should have a return. We start this tail recursion with `None` passed in from the root block. I don't think we can make it non optional, as I don't know what to pass in to start the recursion.

Reply:
You can pass 0, technically. Since you are writing the backend, you should assume that the module is valid.
In a valid module, there is a `return` at the end of a function, including the entry point functions. And with `return` comes `block.termination`, so the `merge_id == 0` would never be used.

Reply:
it's totally up to you though. I'm just pointing out that we are never actually making sense (or use) of `None` here, so we might as well not have it